lib/dependencies/ImportMetaEnvPlugin.js

```diff
@@ -0,0 +1,138 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Natsu @xiaoxiaojx
+*/
+
+"use strict";
+
+const {
+	JAVASCRIPT_MODULE_TYPE_AUTO,
+	JAVASCRIPT_MODULE_TYPE_ESM
+} = require("../ModuleTypeConstants");
+const BasicEvaluatedExpression = require("../javascript/BasicEvaluatedExpression");
+const {
+	evaluateToIdentifier,
+	evaluateToString,
+	toConstantDependency
+} = require("../javascript/JavascriptParserHelpers");
+const ConstDependency = require("./ConstDependency");
+
+/** @typedef {import("../../declarations/WebpackOptions").JavascriptParserOptions} JavascriptParserOptions */
+/** @typedef {import("../Compiler")} Compiler */
+/** @typedef {import("../javascript/JavascriptParser")} JavascriptParser */
+/** @typedef {import("../javascript/JavascriptParser").Range} Range */
+/** @typedef {import("estree").MemberExpression} MemberExpression */
+/** @typedef {import("estree").Identifier} Identifier */
+
+const PLUGIN_NAME = "ImportMetaEnvPlugin";
+class ImportMetaEnvPlugin {
+	/**
+	 * @param {Record<string, EXPECTED_ANY>} definitions environment variable definitions
+	 */
+	constructor(definitions = {}) {
+		this.definitions = definitions;
+	}
+
+	/**
+	 * @param {Compiler} compiler compiler
+	 */
+	apply(compiler) {
+		compiler.hooks.compilation.tap(
+			PLUGIN_NAME,
+			(compilation, { normalModuleFactory }) => {
+				/**
+				 * @param {JavascriptParser} parser parser
+				 * @param {JavascriptParserOptions} parserOptions parser options
+				 * @returns {void}
+				 */
+				const handler = (parser, { importMeta }) => {
+					if (importMeta === false) {
+						return;
+					}
+
+					// import.meta.env
+					parser.hooks.typeof
+						.for("import.meta.env")
```

Comment:
Ideally we should use https://webpack.js.org/configuration/output/#outputimportmetaname, instead `import.meta.env` directly, it is very rare options for very exotic environments, but will be great to support it, so instead directly writing `import.meta` juse use `output.importMetaName`, the same for Environment plugin